docs/apis/subsystems/roles.md

```diff
@@ -0,0 +1,612 @@
+---
+title: Roles API
+tags:
+  - Roles
+  - Interfaces
+  - Capabilities
+---
+## Definitions
```

Comment:
We typically add a brief introduction to the topic at the start of every page rather than jumping into subtitles.

I'd suggest addign a paragraph to the top here to explain what the Roles API is for.